### PR TITLE
(#773) Fix Stylelint Config

### LINKED ISSUE #773: Fix stylelint for ncids-css
In #11 we said we turned on the linter. Technically it is. However, it does not lint any of our files. 
BAD package.json scripts
```
		"lint": "stylelint scss/**/*.scss",
		"lint:fix": "stylelint scss/**/*.scss --fix",
```

GOOD package.json scripts
```
		"lint": "stylelint \"packages/**/*.scss\"",
		"lint:fix": "stylelint \"packages/**/*.scss\" --fix",
```

You will need to go fix all the things too.

All the things:
* file path should be set to packages since uswds 3.0
* wrap glob in escaped quotes
* change stylelint file to js
* keep `selector-class-pattern`, `no-descending-specificity: null`, and `no-duplicate-selectors: null` from [18f](https://www.npmjs.com/package/@18f/identity-stylelint-config?activeTab=code).  see [comment below](https://github.com/NCIOCPL/ncids/issues/773#issuecomment-1808532187)
* remove "ignoreFiles": "./src/stylesheets/lib/**/*"

should be something like this 

```js
module.exports = {
	"extends": [
		"stylelint-config-recommended-scss",
		"stylelint-prettier/recommended"
	],
	"rules": {
		"no-descending-specificity": null,
                "no-duplicate-selectors": null,
    'selector-class-pattern': [
      '^[a-z]([a-z0-9-]+)?(__([a-z0-9]+-?)+)?(--([a-z0-9]+-?)+){0,2}$',
      {
        message:
          'Class selectors should be named using "Two Dashes Style" BEM format. See: https://en.bem.info/methodology/naming-convention/#two-dashes-style',
      },
    ],
	}
}
```
  
once this is complete please see [Enabler: Create Stylelint Rule Package #27](https://github.com/NCIOCPL/cgov-standards-xt/issues/27)

### PR DESCRIPTION
**Pull request by karlikpj**
Closes #773

### CODE REVIEW

**Comment by karlikpj** [comment]
Developer note - don't put comments after selectors. Comments must be above the class block.
```	
/* bad example */
.class {			
  'after', // position
  '05', // margin, units()
  }
```

```	
/* good example */
/* position &   margin, units() */
.class {			
  'after',
  '05',
  }
```

File: packages/ncids-css/.stylelintrc.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+module.exports = {
+	'extends': [
+		'stylelint-config-recommended-scss',
+		'stylelint-prettier/recommended'
+	],
+	'rules': {
+		'unit-no-unknown': null,
+		'no-descending-specificity': null,
+		'no-duplicate-selectors': null,
```
Comment (olitharp-nci) [inline packages/ncids-css/.stylelintrc.js:9]: Remove 
`'unit-no-unknown': null,`
`''no-duplicate-selectors': null,`

Comment (olitharp-nci) [reply packages/ncids-css/.stylelintrc.js:9]: i'll update the ticket here shortly

**Review by olitharp-nci — CHANGES_REQUESTED** [review]

File: packages/ncids-css/packages/nci-autocomplete/internal/_nci-autocomplete.scss [hunk 2]
```diff
@@ -45,7 +46,8 @@
 			background: none;
 		}
 
-		.minPlaceholderMsg {
+		//minPlaceholderMsg
+		.min-placeholder-message {
```
Comment (olitharp-nci) [inline packages/ncids-css/packages/nci-autocomplete/internal/_nci-autocomplete.scss:50]: Please add a note in the commit that you are changing the classname that affects javascript for autocomplete and that you've updated tests that go along with it

**Review by olitharp-nci — CHANGES_REQUESTED** [review]

**Review by olitharp-nci — APPROVED** [review]

**Review by arcepaul — APPROVED** [review]

**Pull request merged**